qiskit/pulse/builder.py

```diff
@@ -0,0 +1,1986 @@
+            One of ``left``, ``right``, ``sequential`` or an alignment
+            contextmanager.
+        default_transpiler_settings: Default settings for the transpiler.
+        default_circuit_scheduler_settings: Default settings for the
+            circuit to pulse scheduler.
+
+    Returns:
+        A new builder context which has the active builder inititalized.
+    """
+    return _PulseBuilder(
+        backend=backend,
+        schedule=schedule,
+        default_alignment=default_alignment,
+        default_transpiler_settings=default_transpiler_settings,
+        default_circuit_scheduler_settings=default_circuit_scheduler_settings)
+
+
+# Builder Utilities
+def _active_builder() -> _PulseBuilder:
+    """Get the active builder in the active context.
+
+    Returns:
+        The active active builder in this context.
+
+    Raises:
+        exceptions.NoActiveBuilder: If a pulse builder function is called
+        outside of a builder context.
+    """
+    try:
+        return BUILDER_CONTEXTVAR.get()
+    except LookupError:
+        raise exceptions.NoActiveBuilder(
+            'A Pulse builder function was called outside of '
+            'a builder context. Try calling within a builder '
+            'context, eg., "with pulse.build() as schedule: ...".')
+
+
+def active_backend():
+    """Get the backend of the currently active builder context.
+
+    Returns:
+        BaseBackend: The active backend in the currently active
+            builder context.
+
+    Raises:
+        exceptions.BackendNotSet: If the builder does not have a backend set.
+    """
+    builder = _active_builder().backend
+    if builder is None:
+        raise exceptions.BackendNotSet(
+            'This function requires the active builder to '
+            'have a "backend" set.')
+    return builder
+
+
+def append_block(block: Schedule):
+    """Call a block by appending to the active builder block."""
+    _active_builder().append_block(block)
+
+
+def append_instruction(instruction: instructions.Instruction):
+    """Append an instruction to the current active builder context block.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+
+        d0 = pulse.DriveChannel(0)
+
+        with pulse.build() as pulse_prog:
+            pulse.append_instruction(pulse.Delay(10, d0))
+
+        print(pulse_prog.instructions)
+    """
+    _active_builder().append_instruction(instruction)
+
+
+def num_qubits() -> int:
+    """Return number of qubits in the currently active backend.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+        from qiskit.test.mock import FakeOpenPulse2Q
+
+        backend = FakeOpenPulse2Q()
+
+        with pulse.build(backend):
+            print(pulse.num_qubits())
+
+    .. note:: Requires the active builder context to have a backend set.
+    """
+    return active_backend().configuration().n_qubits
+
+
+def seconds_to_samples(seconds: float) -> int:
```

Comment:
Can we wait and see here? The error should be very straightforward. I would prefer to keep it this way, so that in the future it might be possible to include this as delayed evaluation at compilation time.